frame/nomination-pools/src/lib.rs

```diff
@@ -1799,67 +1802,117 @@ pub mod pallet {
+					Ok(bonded_pool.bonded_account())
+				},
+				|| {
+					let bonded_pool =
+						BondedPool::<T>::get(pool_id).ok_or(Error::<T>::PoolNotFound)?;
+					Ok(bonded_pool.bonded_account())
+				},
+			)?;
+
+			T::StakingInterface::nominate(bonded_account, validators.clone())?;
+			Self::deposit_event(Event::<T>::Nominated { pool_id, validators });
+
 			Ok(())
 		}
 
+		/// Set the metadata of a pool.
+		///
+		/// Dispatch origin of this call must be signed by the root or state-toggler roles of the
+		/// pool, or be the (holy) root origin.
 		#[pallet::weight(T::WeightInfo::set_state())]
 		pub fn set_state(
 			origin: OriginFor<T>,
 			pool_id: PoolId,
 			state: PoolState,
 		) -> DispatchResult {
-			let who = ensure_signed(origin)?;
-			let mut bonded_pool = BondedPool::<T>::get(pool_id).ok_or(Error::<T>::PoolNotFound)?;
-			ensure!(bonded_pool.state != PoolState::Destroying, Error::<T>::CanNotChangeState);
-
-			if bonded_pool.can_toggle_state(&who) {
-				bonded_pool.set_state(state);
-			} else if bonded_pool.ok_to_be_open(Zero::zero()).is_err() &&
-				state == PoolState::Destroying
-			{
-				// If the pool has bad properties, then anyone can set it as destroying
-				bonded_pool.set_state(PoolState::Destroying);
-			} else {
-				Err(Error::<T>::CanNotChangeState)?;
-			}
+			let bonded_pool = Self::ensure_signed_or_root(
+				origin,
+				|who| {
+					let mut bonded_pool =
+						BondedPool::<T>::get(pool_id).ok_or(Error::<T>::PoolNotFound)?;
+					if bonded_pool.ok_to_be_open(Zero::zero()).is_err() &&
+						state == PoolState::Destroying
+					{
+						// if the pool is in a bad state, we force set it to `Destroying`..
+						bonded_pool.set_state(PoolState::Destroying);
+					} else {
+						// else we set it to whatever the call wanted.
+						ensure!(
+							bonded_pool.state != PoolState::Destroying &&
+								bonded_pool.can_toggle_state(&who),
+							Error::<T>::CanNotChangeState
+						);
+						bonded_pool.set_state(state);
+					}
+					Ok(bonded_pool)
+				},
+				|| {
+					let mut bonded_pool =
+						BondedPool::<T>::get(pool_id).ok_or(Error::<T>::PoolNotFound)?;
+					ensure!(
+						bonded_pool.state != PoolState::Destroying,
+						Error::<T>::CanNotChangeState
+					);
+					bonded_pool.set_state(state);
+					Ok(bonded_pool)
+				},
+			)?;
 
 			bonded_pool.put();
-
 			Ok(())
 		}
 
+		/// Set the metadata of a pool.
+		///
+		/// Dispatch origin of this call must be signed by the root or state-toggler roles of the
+		/// pool, or be the (holy) root origin.
 		#[pallet::weight(T::WeightInfo::set_metadata(metadata.len() as u32))]
 		pub fn set_metadata(
 			origin: OriginFor<T>,
 			pool_id: PoolId,
 			metadata: Vec<u8>,
 		) -> DispatchResult {
-			let who = ensure_signed(origin)?;
+			Self::ensure_signed_or_root(
+				origin,
+				|who| {
+					ensure!(
+						BondedPool::<T>::get(pool_id)
+							.ok_or(Error::<T>::PoolNotFound)?
+							.can_set_metadata(&who),
```

Comment:
This is probably not necessary. Utility has `dispatch_as` which can emulate any user, so we just need a call to make sure that at least sets some user to be able to access functions like this, and then can just assume a signed origin.

Reply:
So, for instance, we can make this the procedure to have governance controlled pools: 

Set your roles all to a funded, key-less account (e.g. council account) and then the governance will then be able to dispatch all of these with `dispatch_as`. 

Other than needing that funded account to exist, it does seem like a nicer idea. That should not be an issue as we can have the treasury itself transfer some funds to a known-key-less account (`0x111..`). I like reusing `dispatch_as` here as well, it seem to have been made exactly for such use cases. 

If you agree with all of this, then PR can be closed.

Reply:
What I am saying is that there should just be a root call which can set the roles from NONE to some arbitrary account, callable by governance, and then that can be combined with `dispatch_as` to access and modify anything without any additional root checks.

Reply:
`update_roles` already supports root, so this is solved.